llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -3240,6 +3277,11 @@ Value *InstCombinerImpl::foldAndOrOfICmps(ICmpInst *LHS, ICmpInst *RHS,
   ICmpInst::Predicate PredL = LHS->getPredicate(), PredR = RHS->getPredicate();
   Value *LHS0 = LHS->getOperand(0), *RHS0 = RHS->getOperand(0);
   Value *LHS1 = LHS->getOperand(1), *RHS1 = RHS->getOperand(1);
+  if (!IsLogical)
+    if (Value *V =
+            foldAndOrOfICmpsWithPow2AndWithZero(Builder, LHS, RHS, IsAnd, Q))
+      return V;
```

Comment:
Why did you add this fold in this position? New exotic folds usually get added to the end, not the start of functions. Especially weird to see this interleaved with a bunch of variable declarations that you aren't using.

Reply:
Moving to the end.